torch/csrc/autograd/python_function.cpp

```diff
@@ -736,6 +732,15 @@ PyObject* process_outputs(PyObject *op_obj, THPFunction* grad_fn, const Unpacked
 
   grad_fn->cdata.num_inputs = num_outputs;
 
+  // Record type, device, and size information about inputs
+  if (grad_fn->cdata.is_executable) {
+    grad_fn->input_info.clear();
```

Comment:
can you do grad_fn->input_info = std::move(unpacked.input_vars) ?

Reply:
No, the types are different: VariableInfo vs Variable